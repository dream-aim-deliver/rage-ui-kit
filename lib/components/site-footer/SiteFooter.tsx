import React from "react";
import { cn } from "@/utils/utils";
import { SatelliteIcon } from "lucide-react"; // Import SatelliteIcon

interface SiteFooterProps {
  brandSection?: React.ReactNode;
  linksSection?: React.ReactNode;
  contactSection?: React.ReactNode;
}

export const SiteFooter: React.FC<SiteFooterProps> = ({
  linksSection,
  contactSection,
}) => {
  return (
    <footer
      className={cn(
        "bg-neutral-100 dark:bg-neutral-800 text-neutral-800 dark:text-white shadow-lg mt-auto",
      )}
    >
      <div className="w-screen px-4 py-6 flex flex-col lg:flex-row justify-between items-center lg:items-start space-y-4 lg:space-y-0 lg:space-x-6">
        {/* Brand Section */}
        <div className="flex items-center gap-2">
          <SatelliteIcon
            size={20}
            className="text-neutral-800 dark:text-white"
          />
          <span className="text-xl font-bold text-neutral-800 dark:text-white">
            SDA
          </span>
          <span className="text-sm text-neutral-500 dark:text-neutral-400">
            Â© 2024
          </span>
        </div>

        {/* Links Section */}
        <nav className="flex flex-wrap justify-center lg:justify-start w-full lg:w-auto text-sm space-x-6 lg:space-x-8">
          {
            linksSection // Here your links will appear
          }
        </nav>

        {/* Contact Section */}
        <div className="text-sm text-center lg:text-left">{contactSection}</div>
      </div>

      {/* Bottom Border for Aesthetic Similarity to Header */}
      <div className="w-full border-t border-neutral-300 dark:border-neutral-700"></div>
    </footer>
  );
};
