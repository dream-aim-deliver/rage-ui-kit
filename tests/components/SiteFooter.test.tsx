import { render, screen } from "@testing-library/react";
import { SiteFooter } from "@/components/site-footer/SiteFooter";
import { expect, describe, it, vi } from "vitest";
import { SVGProps } from "react";

interface SatelliteIconProps extends SVGProps<SVGSVGElement> {}

// Mocking the SatelliteIcon from lucide-react
vi.mock("lucide-react", () => ({
  SatelliteIcon: (props: SatelliteIconProps) => <svg {...props} />,
}));

describe("<SiteFooter />", () => {
  it("should render the SiteFooter with default content", () => {
    render(<SiteFooter />);

    // Check if default branding is rendered
    expect(screen.getByText(/SDA/i)).toBeInTheDocument();
    expect(screen.getByText(/© 2024/i)).toBeInTheDocument();
  });

  it("should render the SiteFooter with provided sections", () => {
    const brandSection = (
      <div className="flex items-center gap-2">
        <span className="text-xl font-bold">SDA</span>
        <span className="text-sm">© 2024</span>
      </div>
    );

    const linksSection = (
      <div>
        <a href="/about">About</a>
        <a href="/privacy">Privacy</a>
      </div>
    );

    const contactSection = (
      <div>
        <a href="mailto:support@customdomain.com">support@customdomain.com</a>
      </div>
    );

    render(
      <SiteFooter
        brandSection={brandSection}
        linksSection={linksSection}
        contactSection={contactSection}
      />,
    );

    // Check if custom branding is rendered
    expect(screen.getByText(/SDA/i)).toBeInTheDocument();
    expect(screen.getByText(/© 2024/i)).toBeInTheDocument();

    // Check if custom links are rendered
    expect(screen.getByText(/About/i)).toBeInTheDocument();
    expect(screen.getByText(/Privacy/i)).toBeInTheDocument();

    // Check if custom contact is rendered
    expect(screen.getByText(/support@customdomain.com/i)).toBeInTheDocument();
  });

  it("should match the snapshot", () => {
    const { container } = render(<SiteFooter />);
    expect(container).toMatchSnapshot();
  });
});
