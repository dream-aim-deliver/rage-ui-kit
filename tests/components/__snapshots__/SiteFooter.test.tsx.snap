// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`<SiteFooter /> > should match the snapshot 1`] = `
<div>
  <footer
    class="bg-neutral-100 dark:bg-neutral-800 text-neutral-800 dark:text-white shadow-lg mt-auto"
  >
    <div
      class="w-screen px-4 py-6 flex flex-col lg:flex-row justify-between items-center lg:items-start space-y-4 lg:space-y-0 lg:space-x-6"
    >
      <div
        class="flex items-center gap-2"
      >
        <svg
          class="text-neutral-800 dark:text-white"
          size="20"
        />
        <span
          class="text-xl font-bold text-neutral-800 dark:text-white"
        >
          SDA
        </span>
        <span
          class="text-sm text-neutral-500 dark:text-neutral-400"
        >
          Â© 2024
        </span>
      </div>
      <nav
        class="flex flex-wrap justify-center lg:justify-start w-full lg:w-auto text-sm space-x-6 lg:space-x-8"
      />
      <div
        class="text-sm text-center lg:text-left"
      />
    </div>
    <div
      class="w-full border-t border-neutral-300 dark:border-neutral-700"
    />
  </footer>
</div>
`;
